app/grandchallenge/workstations/static/workstations/js/session-control.js

```diff
@@ -48,11 +49,15 @@ function openWorkstationSession(element) {
     }
 }
 
-function getSessionOrigins(hostname, regions) {
-    const protocol = "https";
+function getSessionOrigins(hostname, regions, debug) {
+    const protocol = window.location.protocol;
     const port = window.location.port;
 
-    return regions.map((region) => `${protocol}://${region}.${hostname}${port ? ':' + port : ''}`);
+    if (debug){
```

Comment:
But then we'd be matching on the domain name again, right?

Reply:
No, change the `workstation_regions` context processor to do something like:

```
def workstation_domains(*_, **__):
    return [
      *["https://{region}.domain" for region in settings.WORKSTATIONS_ACTIVE_REGIONS],
      *settings.EXTRA_WORKSTATION_BROADCAST_DOMAINS 
    ]
```

Then settings.py

```
EXTRA_WORKSTATION_BROADCAST_DOMAINS = []
```

Then in your test

```
def test_blah(settings, live_server, ...):
   settings.EXTRA_WORKSTATION_BROADCAST_DOMAINS = [live_server.url]
```

Then update your JS to rather than take a list of subdomains that you then construct a list of URLs from to just take a list of URLs.

Reply:
Ah this is very elegant. I had misunderstood your previous comment. This works like a charm! Thanks!